lib/_http_client.js

```diff
@@ -692,7 +692,14 @@ function parserOnIncomingClient(res, shouldKeepAlive) {
     // We already have a response object, this means the server
     // sent a double response.
     socket.destroy();
-    return 0;  // No special treatment.
+    if (socket.parser) {
+      // https://github.com/nodejs/node/issues/60025
+      // Now, parser.incoming is pointed to the new IncomingMessage,
+      // we need to rewrite it to the first one and skip all the pending IncomingMessage
+      socket.parser.incoming = req.res;
+      socket.parser.incoming.skipPendingData = true;
```

Comment:
I think this should probably be a `kSkipPendingData` symbol, just to make sure this doesn't accidentally become a public API at some point.